bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/internal/IdentProtocol.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.irobot.internal;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.nio.charset.StandardCharsets;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import com.google.gson.stream.JsonReader;
+
+/**
+ * iRobot discovery and identification protocol
+ *
+ * @author Pavel Fedin <pavel_fedin@mail.ru> - Initial contribution
+ *
+ */
+public class IdentProtocol {
+
+    private static final String UDP_PACKET_CONTENTS = "irobotmcs";
+    private static final int REMOTE_UDP_PORT = 5678;
+
+    public static DatagramSocket sendRequest(InetAddress host) throws IOException {
+        DatagramSocket socket = new DatagramSocket();
+
+        socket.setBroadcast(true);
+        socket.setReuseAddress(true);
+
+        byte[] packetContents = UDP_PACKET_CONTENTS.getBytes(StandardCharsets.UTF_8);
+        DatagramPacket packet = new DatagramPacket(packetContents, packetContents.length, host, REMOTE_UDP_PORT);
+
+        socket.send(packet);
+        return socket;
+    }
+
+    public static DatagramPacket receiveResponse(DatagramSocket socket) throws IOException {
+        byte[] buffer = new byte[1024];
+        DatagramPacket incomingPacket = new DatagramPacket(buffer, buffer.length);
+
+        socket.setSoTimeout(1000 /* one second */);
+        socket.receive(incomingPacket);
+
+        return incomingPacket;
+    }
+
+    public static IdentData decodeResponse(DatagramPacket packet) throws JsonParseException, ClassCastException {
```

Comment:
What is throwing a `ClassCastException` here?

Reply:
Old version of the parser, which doesn't exist any more. :) Thanks for pointing out.